### PR TITLE
Release notes august 2022

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

File: docs/rule-updates.mdx [hunk 1]
```diff
@@ -12,6 +12,116 @@ import MoreHelp from "/src/components/MoreHelp"
 
 Welcome to monthly rule updates! This document includes selected new rules, removed false positives (FP), and other rule updates. These new rules and their updates are made by the Semgrep community and r2c.
 
+## August 2022
```
Comment (adamkvitek) [inline docs/rule-updates.mdx:15]: @minusworld and @raghavjain3 please check whether the structure of this update looks fine - feel free to suggest edits or commit anything you need.

Comment (minusworld) [reply docs/rule-updates.mdx:15]: Seems OK to me. Thanks! :D

File: docs/release-notes.md [hunk 2]
```diff
@@ -8,7 +8,42 @@ toc_max_heading_level: 2
...
+
+#### Changes
+
+- Previously, the following error message appeared when metrics have not been uploaded within the set timeout timeframe:
+  ```
+  Error in send: HTTPSConnectionPool(host='metrics.semgrep.dev', port=443): Read timed out. (read timeout=3)
+  ```
+  As this caused confusion when running the CLI, this message is now displayed for development and debugging purposes only. Note that metrics are still successfully uploaded, but the success status is not sent in time for the current timeout set.
+
+- `semgrep ci` now defaults to fail open and always exits with exit code 0, which is equivalent to passing `--suppress-errors`. To disable this behavior, you can pass `--no-suppress-errors`, surfacing all exit codes to the CI provider. See [Configuring blocking findings and errors](/semgrep-ci/configuration-reference.md/#configuring-blocking-findings-and-errors) for more information.
```
Comment (s-santillan) [inline docs/release-notes.md:42]:
```suggestion
- `semgrep ci` now defaults to fail open on internal errors and always exits with exit code 0, which is equivalent to passing `--suppress-errors`. To disable this behavior, you can pass `--no-suppress-errors`, surfacing all exit codes to the CI provider. See [Configuring blocking findings and errors](/semgrep-ci/configuration-reference.md/#configuring-blocking-findings-and-errors) for more information.
```

Comment (s-santillan) [reply docs/release-notes.md:42]: There are two possible fail opens in Semgrep, (fail open on findings, fail open on internal errors) so I just specified which one.

**Review by s-santillan — APPROVED** [review]
lgtm with 1 suggested comment.

File: docs/rule-updates.mdx [hunk 3]
```diff
@@ -12,6 +12,116 @@ import MoreHelp from "/src/components/MoreHelp"
...
+
+### Community tier
+
+#### New rules from Segmrep community and r2c
```
Comment (raghavjain3) [inline docs/rule-updates.mdx:19]:
```suggestion
#### New rules from Semgrep community and r2c
```

**Review by raghavjain3 — APPROVED** [review]

File: docs/release-notes.md (hunk 2 above)
Comment (adamkvitek) [reply docs/release-notes.md:42]: Yeah, I think the term fail open is confusing by itself.

File: docs/release-notes.md [hunk 4]
```diff
@@ -8,7 +8,42 @@ toc_max_heading_level: 2
...
+### Semgrep CLI
+
+These release notes include upgrades for versions ranging between 0.108.0 and 0.111.0.
+
+#### Additions
+
+- Semgrep now provides experimental support for the **Swift** language. See all languages that Semgrep supports in [Supported languages](https://semgrep.dev/docs/supported-languages/).
+- Add configuration options for using the tree-sitter library installed anywhere on the system.
+- Metrics now include language-aggregated parse rates (files, bytes). The purpose of this is to continue with parsing improvements. See [Semgrep privacy policy](/metrics.md) for more details.
+- Semgrep CI now accepts more formats of Git URLs for metadata that are sent to semgrep.dev.
```
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/release-notes.md:32]: @bmahe - I am removing them based on a comment from @s-santillan - these variables are now buggy - see https://returntocorp.slack.com/archives/C02PV1VVAUR/p1659544394409909

[the remark this answers is not in the document]
Comment (milanwilliams) [reply docs/release-notes.md:32]: @adamkvitek Could we still document this feature, but clarify that it's experimental perhaps with a "beta" flag? It is really useful for folks - this is how they can get working links on the Findings page. But we can explain that it's still being worked on & we appreciate any bug reports / suggestions!

[the remark this answers is not in the document]
Comment (s-santillan) [reply docs/release-notes.md:32]: I am okay with keeping it.

[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/release-notes.md:32]: @milanwilliams I hope I got it correctly, the text is below:

```suggestion
- Semgrep CI now accepts more formats of Git URLs for metadata that are sent to semgrep.dev. This work-in-progress functionality is to enable working links from the Semgrep App Findings page. The user provides a fallback for repository name (SEMGREP_REPO_NAME) and repository URL (SEMGREP_REPO_URL) if these values are undefined by the CI job. We appreciate any bug reports or suggestions as this feature is still in development.
```

[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/release-notes.md]: Note: These variables can become stable soon. They are not experimental also.

[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/release-notes.md]: @milanwilliams please, what has a priority - if a CI job has these variables defined (`SEMGREP_REPO_NAME` and `SEMGREP_REPO_URL`) - does it override the user setup of these variables (when a user defines them)?

**Pull request merged**